pkg/stanza/fileconsumer/matcher/internal/filter/filter.go

```diff
@@ -39,6 +40,61 @@ func Filter(values []string, regex *regexp.Regexp, opts ...Option) ([]string, er
 	return result, errs
 }
 
+func FilterFiles(values []string, includeregex []*regexp.Regexp, excluderegex []*regexp.Regexp) ([]string, error) {
+
+	result := []string{}
+	var errs error
+
+	for _, value := range values {
+
+		parts := strings.Split(value, "/")
+		parts = strings.SplitN(parts[4], "_", 2)
```

Comment:
You cannot hard code "4" like this.
These are generic code. Its not used only for your pod logs. It can be used in any receiver.